catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt

```diff
@@ -0,0 +1,104 @@
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.Script
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.content.StartScript
+import org.catrobat.catroid.content.bricks.ChangeYByNBrick
+import org.catrobat.catroid.content.bricks.FormulaBrick
+import org.catrobat.catroid.content.bricks.RepeatBrick
+import org.catrobat.catroid.content.eventids.EventId
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.FormulaElement
+import org.catrobat.catroid.formulaeditor.Sensors
+import org.catrobat.catroid.test.MockUtil
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class RepeatActionTest(
+    private var iterations: Int,
+    private var deltaY: Int,
+    private var loopCondition: Formula?,
+    private var beforeLoopBrick: FormulaBrick?,
+    private var afterLoopBrick: FormulaBrick?,
+    private var expectedValue: Int
+) {
+    private lateinit var sprite: Sprite
+    private lateinit var project: Project
+    private lateinit var script: Script
+
+    private val delayByContract = 0.020f
+    private val deltaDelayByContract = 0.005f
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{index}")
+        fun parameters() = listOf(
+            arrayOf(4, -10, Formula(4), null, ChangeYByNBrick(50), -40),
+            arrayOf(8, -10, Formula(FormulaElement(FormulaElement.ElementType.SENSOR, Sensors.OBJECT_Y.name, null)),
+                    ChangeYByNBrick(5), ChangeYByNBrick(50), 5),
+            arrayOf(4, -1, Formula(-1), null, ChangeYByNBrick(10), 10),
+            arrayOf(4, -1, Formula(0), null, ChangeYByNBrick(10), 10),
+            arrayOf(4, -1, Formula("3"), null, ChangeYByNBrick(10), 7),
+            arrayOf(4, -1, Formula(Double.NaN), null, ChangeYByNBrick(10), 10))
+    }
+
+    @Before
+    fun setUp() {
+        sprite = Sprite("testSprite")
+        script = StartScript()
+        project = Project(MockUtil.mockContextForProject(), "testProject")
+        ProjectManager.getInstance().currentProject = project
+    }
+
+    @Test
+    fun testLoopDelay() {
+        val repeatBrick = RepeatBrick(loopCondition)
+        repeatBrick.addBrick(ChangeYByNBrick(deltaY))
+        if (beforeLoopBrick != null) {
+            script.addBrick(beforeLoopBrick)
+        }
+        script.addBrick(repeatBrick)
+        if (afterLoopBrick != null) {
+            script.addBrick(afterLoopBrick)
+        }
+        sprite.addScript(script)
+        sprite.initializeEventThreads(EventId.START)
+        for (index in 0 until iterations) {
+            var time = 0.0
+            println("index: $index")
+            while (time < delayByContract) {
+                sprite.look.act(deltaDelayByContract)
+                time += deltaDelayByContract.toDouble()
+            }
+        }
```

Comment:
Please use Kotlin functions instead and also don't push debug lines.

```suggestion
        repeat((iterations * delayByContract / deltaDelayByContract).toInt()){ sprite.look.act(deltaDelayByContract) }
```